content/assignments/parallel-graph/skel/os_threadpool.h

```diff
@@ -0,0 +1,33 @@
+#ifndef __SO_THREADPOOL_H__
+#define __SO_THREADPOOL_H__
+
+#include <pthread.h>
+
+typedef struct {
+    void *argument;
+    void (*task)(void *);
+} os_task_t;
+
+typedef struct _node{
+    os_task_t *task;
+    struct _node *next;
+} os_task_queue_t;
+
+typedef struct {
+    unsigned int should_stop;
+
+    unsigned int num_threads;
+    pthread_t *threads;
+
+    os_task_queue_t *tasks;
+} os_threadpool_t;
+
+os_task_t *task_create(void *arg, void (*f)(void *));
+void add_task_in_queue(os_threadpool_t *tp, os_task_t *t);
+os_task_t *get_task(os_threadpool_t *tp);
+os_threadpool_t *_os_threadpool_create();
```

Comment:
How should we use this function? Seems to be something that should be used only in `os_threadpool.c`.

Reply:
Yes. It should be a wrapper over malloc